Library/Homebrew/cmd/update.sh

```diff
@@ -331,6 +345,13 @@ EOS
     UPSTREAM_BRANCH="$(upstream_branch)"
     # the refspec ensures that the default upstream branch gets updated
     (
+      if [[ -n "$HOMEBREW_UPDATE_PREINSTALL" ]]
+      then
+        # Skip taps without formulae.
+        FORMULAE="$(find "$DIR" -maxdepth 1 -name '*.rb' -or -name 'Formula' -print -quit)"
```

Comment:
The `find` invocation actually needs to look as follows, otherwise it will fail for taps with formulae in the repository root (like `homebrew/versions`):

```sh
find "$DIR" -maxdepth 1 \( -name '*.rb' -or -name 'Formula' \) -print -quit
```

This is due to the implicit `-and` and the precedence relationship between those arguments.